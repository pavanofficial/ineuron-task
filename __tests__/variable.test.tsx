import { render, screen, fireEvent } from '@testing-library/react';
import "@testing-library/jest-dom";
import mockRouter from 'next-router-mock';
import Variable from '../src/pages/variable/[...slug]'
jest.mock('next/router', () => require('next-router-mock'));


it('renders homepage unchanged', () => {
  const { container } = render(<Variable />)
  expect(container).toMatchSnapshot()
})

describe('next-router-mock', () => {
    it('mocks the useRouter hook', () => {
      mockRouter.push("/$1/0/2");
      
      // Ensure the router was updated:
      expect(mockRouter).toMatchObject({ 
        asPath: "/$1/0/2",
        pathname: "/$1/0/2",
        query: { },
      })
    })
})

describe("Details", () => {
    it("renders the criteria listing", () => {
      render(<Variable />);
      expect(screen.getByTestId("backBtn")).toBeInTheDocument();
      expect(screen.getByTestId("contentBox")).toBeInTheDocument();
      expect(screen.getByTestId("title")).toBeInTheDocument();
    })
})