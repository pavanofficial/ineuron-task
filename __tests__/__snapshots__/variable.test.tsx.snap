// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <main
    class="flex flex-col justify-center items-center p-6 bg-gray-100 min-h-[100vh]"
  >
    <div>
      <button
        class="inline-flex gap-2 items-center my-3 text-black"
        data-testid="backBtn"
      >
        <svg
          class="w-5 h-5"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10.5 19.5L3 12m0 0l7.5-7.5M3 12h18"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        Go back
      </button>
      <div
        class="overflow-hidden bg-white shadow sm:rounded-md md:min-w-[30vw] px-6 py-5 sm:px-6 w-4ull"
        data-testid="contentBox"
      >
        <h3
          class="text-2xl font-medium leading-6 text-gray-900"
          data-testid="title"
        >
          Variable params
        </h3>
        <hr
          class="w-full border-[0.1px] border-gray-200 mt-5"
        />
        <ul
          class="divide-y divide-gray-200"
          role="list"
        />
      </div>
    </div>
  </main>
</div>
`;
